File: srm177/stairs.py
class Stairs:
	def designs(self, maxHeight, minWidth, totalHeight, totalWidth):
		designs = 0
		for i in range(maxHeight, 0, -1):
			if totalHeight % i == 0 and totalHeight != i:
				if (totalWidth % (totalHeight / i - 1) == 0) and (totalWidth / (totalHeight / i - 1)) >= minWidth:
					designs += 1
					
		return designs

File: srm177/Stairs.java
public class Stairs {
	
	public int designs(int maxHeight, int minWidth, int totalHeight, int totalWidth) {
		int design = 0;
		for (int i = maxHeight; i > 0; i--) {
			if (totalHeight % i == 0 && totalHeight != i) {
				if ((totalWidth % (totalHeight / i - 1) == 0) && (totalWidth / (totalHeight / i - 1)) >= minWidth) {
					design += 1;
				}
			}
		}			
		return design;
	}
}